core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -135,9 +139,27 @@ private static XContentBuilder build(RestChannel channel, RestStatus status, Exc
         }
 
         XContentBuilder builder = channel.newErrorBuilder().startObject();
+        builder.field(STATUS, status.getStatus());
         ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
-        builder.field("status", status.getStatus());
         builder.endObject();
         return builder;
     }
+
+    public static ElasticsearchException errorFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        ElasticsearchException exception = null;
+
+        String currentFieldName = parser.currentName();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            }
+            if (STATUS.equals(currentFieldName) == false) {
+                exception = ElasticsearchException.failureFromXContent(parser);
+            }
```

Comment:
It won't be the same, actually I think the status after parsing will always be INTERNAL_SERVER_ERROR. When rendered without details, the cause of the exception is lost so there's no way to retrieve it or to derive a status code from it. But even when the exception is rendered with detail, since we lost the type after parsing it looks like it will always produce an INTERNAL_SERVER_ERROR.

If we want to keep the status code after parsing, I think it should be rendered at the same level as `type/reason/cause` in the ElasticsearchException.innerToXContent() method.